static/app/views/issueDetails/groupActivity.tsx

```diff
@@ -1,219 +1,111 @@
-  inputId: string;
-};
-
-class GroupActivity extends Component<Props, State> {
-  // TODO(dcramer): only re-render on group/activity change
-  state: State = {
-    createBusy: false,
-    error: false,
-    errorJSON: null,
-    inputId: uniqueId(),
-  };
-
-  handleNoteDelete = async ({noteId, text: oldText}) => {
-    const {api, group, organization} = this.props;
-
-    addLoadingMessage(t('Removing comment\u{2026}'));
-
-    try {
-      await deleteNote(api, organization.slug, group, noteId, oldText);
-      clearIndicators();
-    } catch (_err) {
+function GroupActivity(props: Props) {
+  const organization = useOrganization();
+  const {group} = props;
+  const {activity: activities, count, id: groupId} = group;
+  const groupCount = Number(count);
+  const mostRecentActivity = getGroupMostRecentActivity(activities);
+  const reprocessingStatus = getGroupReprocessingStatus(group, mostRecentActivity);
+  const mutators = useMutateActivity({
+    organization,
+    group,
+  });
+
+  const deleteMutationOptions: MutateOptions<TData, TError, TVariables, TContext> = {
+    onError: () => {
       addErrorMessage(t('Failed to delete comment'));
-    }
+    },
+    onSuccess: () => {
+      addSuccessMessage(t('Comment removed'));
+    },
   };
 
-  /**
-   * Note: This is nearly the same logic as `app/views/alerts/details/activity`
-   * This can be abstracted a bit if we create more objects that can have activities
-   */
-  handleNoteCreate = async note => {
-    const {api, group, organization} = this.props;
-
-    this.setState({
-      createBusy: true,
-    });
-
-    addLoadingMessage(t('Posting comment\u{2026}'));
-
-    try {
-      await createNote(api, organization.slug, group, note);
-
-      this.setState({
-        createBusy: false,
-
-        // This is used as a `key` to Note Input so that after successful post
-        // we reset the value of the input
-        inputId: uniqueId(),
-      });
-      clearIndicators();
-    } catch (error) {
-      this.setState({
-        createBusy: false,
-        error: true,
-        errorJSON: error.responseJSON || DEFAULT_ERROR_JSON,
-      });
+  const addMutationOptions: MutateOptions<TData, TError, TVariables, TContext> = {
+    onError: () => {
       addErrorMessage(t('Unable to post comment'));
-    }
+    },
+    onSuccess: data => {
+      GroupStore.addActivity(group.id, data);
+      addSuccessMessage(t('Comment posted'));
+    },
   };
 
-  handleNoteUpdate = async (note, {noteId, text: oldText}) => {
-    const {api, group, organization} = this.props;
-
-    addLoadingMessage(t('Updating comment\u{2026}'));
-
-    try {
-      await updateNote(api, organization.slug, group, note, noteId, oldText);
-      clearIndicators();
-    } catch (error) {
-      this.setState({
-        error: true,
-        errorJSON: error.responseJSON || DEFAULT_ERROR_JSON,
-      });
+  const updateMutationOptions: MutateOptions<TData, TError, TVariables, TContext> = {
+    onError: () => {
```

Comment:
there's something funky about passing these around along with the mutators, maybe handle the entire mutation in GroupActivity (this file) and instead add a onCreate, onEdit, onDelete to the section? Feels like the event is handled in two places